packages/reporting/src/patterns.js

```diff
@@ -126,6 +141,53 @@ const TRANSFORMS = new Map(
       }
     },
 
+    /**
+     * Given a URL and a list of query parameters, it returns an equivalent
+     * URL, but with those query parameters removed.
+     *
+     * Notes:
+     * - If the parameter occurs multiple times, all of them will be removed.
+     * - If the URL is invalid, null is returned.
+     *
+     * Example ["removeParams", ["foo"]]:
+     * - "https://example.test/path?foo=remove&bar=keep" -> "https://example.test/path?bar=keep"
+     * - "This is a string but not an URL" -> null
+     * - "/example.test/path" -> null (relative URLs are not supported)
+     *
+     * Example ["removeParams", ["foo", "bar"]]:
+     * - "https://example.test/path?foo=1&bar=2" -> "https://example.test/path"
+     */
+    removeParams: (x, queryParams) => {
+      expectString(x);
+      expectArrayOfStrings(queryParams);
+      try {
+        const url = new URL(x);
+
+        // Why not url.searchParams? The parsing step in url.searchParams
+        // loses information (see https://stackoverflow.com/a/45516812/783510).
+        if (!url.search) {
+          return x;
+        }
+        const parts = url.search
+          .slice(1)
+          .split('&')
+          .filter((x) => !queryParams.includes(split0(x, '=')));
```

Comment:
please rename `x` - it is used as argument to removeParams already. calling it a `param` would probably make sense